### PR TITLE
adding check and error message for empty workspace on create vdb

### PR DESCRIPTION
**Pull request by bfitzpat**
Signed-off-by: bfitzpat@redhat.com <bfitzpat@redhat.com>

### CODE REVIEW

**Comment by bfitzpat** [comment]
image(1)

**Comment by bfitzpat** [comment]
image

**Comment by bfitzpat** [comment]
Though that *will* work, I have some concerns about that approach. The user will need to then activate the extension, which requires even another step. Perhaps we build in a test to disable menu/toolbar buttons if the workspace isn't ready. You can also add a workspace listener to update the state of those menu/toolbar items. 

WDYT?

**Comment by bfitzpat** [comment]
Updated the PR slightly to be more in line with the early check and workspace listener.

File: src/extension.ts [hunk 1]
```diff
@@ -50,9 +52,21 @@ export function activate(context: vscode.ExtensionContext) {
...
 	DATASOURCE_TYPES.set('Rest Based', new RestBasedDataSource(TEMPLATE_NAME));
 
+	if (!vscode.workspace.workspaceFolders || vscode.workspace.workspaceFolders.length === 0) {
+		workspaceReady = false;
+	}
+
 	dataVirtProvider = new DataVirtNodeProvider(vscode.workspace.rootPath, context);
 	creatDataVirtView();
 
+	vscode.workspace.onDidChangeWorkspaceFolders( event => {
```
Comment (apupier) [inline src/extension.ts]: as event is unused, we can replace with () as suggested by quickix in VS Code

Comment (bfitzpat) [reply src/extension.ts]: Fixed

**Pull request merged**